ie/apply-action.go

```diff
@@ -47,30 +50,88 @@ func (i *IE) ApplyAction() ([]byte, error) {
 	}
 }
 
+// Returns true when Apply Action flags are set in an inconsistent way
+func (i *IE) inconsistentApplyActionFlags() (malformed bool) {
+	// One and only one of the DROP, FORW, BUFF, IPMA and IPMD flags shall be set to "1".
+	flags := []bool{i.HasDROP(), i.HasFORW(), i.HasBUFF(), i.HasIPMA(), i.HasIPMD()}
+	counter := 0
+	for _, v := range flags {
+		if v {
+			counter++
+		}
+	}
+	if counter != 1 {
+		return true
+	}
+	// The NOCP flag and BDPN flag may only be set if the BUFF flag is set.
+	if (i.HasNOCP() || i.HasBDPN()) && !i.HasBUFF() {
+		return true
+	}
+	// The DUPL flag may be set with any of the DROP, FORW, BUFF and NOCP flags.
+	if i.HasDUPL() && !(i.HasDROP() || i.HasFORW() || i.HasBUFF() || i.HasNOCP()) {
+		return true
+	}
+	// The DFRT flag may only be set if the FORW flag is set.
+	// Note: in TS 29.244 V18.0.1 (most recent as of writing), there is a typo and DFRN is stated instead of DFRT
+	if i.HasDFRT() && !i.HasFORW() {
+		return true
+	}
+	// The DDPN flag may be set with any of the DROP and BUFF flags.
+	if i.HasDDPN() && !(i.HasDROP() || i.HasBUFF()) {
+		return true
+	}
+
+	// Note: The following is also stated in TS 29.244 section 8.2.26 V18.0.1,
+	// but since "may" is used and not "may only"
+	// it cannot be used to check for inconsistent IEs:
+	// - The EDRT flag may be set if the FORW flag is set.
+	// - Both the MBSU flag and the FSSM flag may be set […]
+	return false
+}
+
 // HasDROP reports whether an IE has DROP bit.
 func (i *IE) HasDROP() bool {
-	if len(i.Payload) < 1 {
-		return false
-	}
 
 	switch i.Type {
 	case ApplyAction, DataStatus:
+		if len(i.Payload) < 1 {
+			return false
+		}
 		return has1stBit(i.Payload[0])
 	default:
+		v, err := i.ApplyAction()
+		if err == nil {
+			if len(v) < 1 {
```

Comment:
The `v` length < 1 check is no more needed because already checked in `i.ApplyAction()`.

Reply:
Thanks, I removed it in bc1c877.